pkg/server/server_controller_new_server.go

```diff
@@ -246,6 +246,8 @@ func makeSharedProcessTenantServerConfig(
 	baseCfg.SpanConfigsDisabled = kvServerCfg.BaseConfig.SpanConfigsDisabled
 	baseCfg.EnableDemoLoginEndpoint = kvServerCfg.BaseConfig.EnableDemoLoginEndpoint
 	baseCfg.DefaultZoneConfig = kvServerCfg.BaseConfig.DefaultZoneConfig
+	baseCfg.HeapProfileDirName = kvServerCfg.BaseConfig.HeapProfileDirName
```

Comment:
nit: does this impact tests? Or is it just some cleanup along the way.

Reply:
This is exercised in tests; its lack was noticed by a failure in #110001.